### ISSUE TITLE
getSessionFileList does not call back after failed source sync

### ISSUE THREAD

**Issue by harunsmrkovic** [digime/digime-sdk-android#35]
## Checklist
- [ ] I’m using the latest version of the digi.me Private Sharing SDK.
- [x] I'm using the latest version of the digi.me App.
- [x] I've searched for existing [GitHub issues](https://github.com/digime/digime-sdk-android/issues)
- [x] I've looked for existing answers in the [SDK Documentation](https://digime.github.io/digime-sdk-android/index.html) and [Developer FAQs](https://digime.freshdesk.com/support/solutions/9000115894) 
- [x] This issue does not relate to security and can be disclosed publicly on Github. 
## Environment
Describe your development environment here. Please provide as much information as you can, noting the following details if possible:
- Kotlin Version: `X.X.X`
- Android SDK Version: `3.0.1-ubdi`
- Installation Platform & Version: `[Gradle/Maven]` version `X.X.X`
## Goals
What are you trying to achieve? 
Get updates via callback
## Expected Results
What do you expect to happen?
Get updates via callback
## Actual Results
What actually happened?
Do not get updates via callback, if some of the sources have failed in previous sync
## Steps to Reproduce
What are the steps necessary to reproduce this issue?
Try to do a sync through UBDI app (use re-sync button at bottom of Data Sources screen), but make sure that some of your sources will fail this sync (revoke auth from the source?)
Then, after it has failed, try to do Re-Sync again. Sources will remain in the `pending` state forever.
This is indicative of the callback in `getSessionFileList` not being called.

**Comment by JacobKingDev** [comment]
@harunsmrkovic Do you still see this happening? We haven't been able to reproduce, so will close if you don't see it anymore?

**Comment by harunsmrkovic** [comment]
We are still able to reproduce this, but it is not limited to previously failed source sync. It just randomly sometimes does not work.

Now, I have reproduced it from the first try, in a clean UBDI app build. App is logging stuff as
```
(Console Logs) 12-02 19:52:52.096  D/DigiMeSDK(24538): Session data poll scheduled.
(Console Logs) 12-02 19:52:52.096  I/DigiMeSDK(24538): Sync still in progress, continuing to poll for updates.
```

a ton of times, and updates are obviously happening, but callback handler is not being called.

Also, in last instance, I've called `getSessionData`.
